In [2]:
import numpy as np
from scipy import stats

In [3]:
sample = [172,174, 168, 169, 171,173, 175, 170, 169, 172]
sample_mean= np.mean(sample)
sample_std=np.std(sample, ddof=1) # in t test ddof is always 1 
mean_population= 170
n= len(sample)

print(sample_mean)

171.3


In [ ]:
t_stats=(sample_mean - mean_population) / (sample_std / np.sqrt(n))
t_stats  #1.77

np.float64(1.7782469350914734)

In [5]:
p_value = 2* (1-stats.t.cdf(abs(t_stats), df = n-1))

alpha=0.05

In [ ]:
# Ulta than critical value

if p_value< alpha:
    print("We will reject the null hyp")
else: 
    print("Accept the null hypothesis")

Accept the null hypothesis


# T-Test

## What is a T-Test?

A T-test checks whether the difference between a **sample mean** and a **population mean** is statistically significant.

Question it answers:

> "Is my sample mean different because of random chance, or is the actual population mean different?"

---

## Why do we use a T-Test?

Use a T-test when:

- Population standard deviation (σ) is **unknown**
- Sample size is usually **small (n < 30)**

Since we don't know the actual population spread, we estimate it using the sample's standard deviation.

Because of this extra uncertainty, we use the **t-distribution** instead of the normal (z) distribution.

---

# Z-Test vs T-Test

| Z-Test | T-Test |
|--------|---------|
| Population σ is known | Population σ is unknown |
| Uses Normal Distribution | Uses T Distribution |
| Uses Z-score | Uses T-statistic |
| Fixed shape | Shape depends on degrees of freedom |

---

# Formula

### T-statistic

\[
t=\frac{\bar{x}-\mu}{s/\sqrt{n}}
\]

where,

- \(\bar{x}\) = sample mean
- \(\mu\) = population mean
- \(s\) = sample standard deviation
- \(n\) = sample size

---

# Your Example

Sample:

```python
[172,174,168,169,171,173,175,170,169,172]
```

Population mean:

```python
μ = 170
```

Sample mean:

```text
x̄ = 171.3
```

Sample standard deviation:

```text
s ≈ 2.311
```

Sample size:

```text
n = 10
```

---

# Step 1

Difference between means:

```text
171.3 - 170 = 1.3
```

---

# Step 2

Standard Error:

```text
SE = s / √n
   = 2.311 / √10
   = 2.311 / 3.162
   ≈ 0.7307
```

Standard Error means:

> How much sample means usually vary due to random sampling.

---

# Step 3

Calculate t-statistic:

```text
t = 1.3 / 0.7307
  ≈ 1.778
```

Meaning:

> Your sample mean is 1.778 standard errors above the population mean.

---

# Degrees of Freedom (df)

For a one-sample t-test:

```text
df = n - 1
```

Here,

```text
df = 10 - 1 = 9
```

Degrees of freedom decide the exact shape of the t-distribution.

---

# Paper Method (Critical Value)

Given:

```text
α = 0.05
df = 9
```

From the t-table:

```text
tcritical = ±2.262
```

Decision:

```text
If |t| > 2.262
        Reject H₀
Else
        Fail to Reject H₀
```

Your value:

```text
|1.778| = 1.778
```

Since:

```text
1.778 < 2.262
```

### Conclusion

```text
Fail to Reject H₀
```

There is not enough evidence to say that the population mean is different from 170.

---

# P-value Method

Your code calculates:

```text
p ≈ 0.109
```

Interpretation:

> Assuming the true mean is 170, there is about a 10.9% chance of getting a sample this extreme (or more extreme).

Decision:

```text
p = 0.109
α = 0.05
```

Since:

```text
0.109 > 0.05
```

### Conclusion

```text
Fail to Reject H₀
```

---

# Code Explanation

### 1.

```python
sample_mean = np.mean(sample)
```

Calculates:

```text
x̄ = Σx / n
```

Returns the sample average.

---

### 2.

```python
sample_std = np.std(sample, ddof=1)
```

Calculates sample standard deviation.

`ddof=1` means divide by:

```text
n - 1
```

instead of

```text
n
```

because population σ is unknown and we are estimating it from the sample.

---

### 3.

```python
n = len(sample)
```

Returns the number of observations.

Here:

```text
n = 10
```

---

### 4.

```python
t_stats = (sample_mean - mean_population) / (sample_std / np.sqrt(n))
```

Calculates:

\[
t=\frac{\bar{x}-\mu}{s/\sqrt{n}}
\]

- Numerator → difference between means
- Denominator → standard error

---

### 5.

```python
np.sqrt(n)
```

Calculates:

```text
√n
```

Used in the standard error formula.

---

### 6.

```python
stats.t.cdf(abs(t_stats), df=n-1)
```

- `abs()` → makes t positive
- `df=n-1` → degrees of freedom
- `cdf()` → cumulative probability to the left of that t-value

---

### 7.

```python
1 - stats.t.cdf(...)
```

Gives the probability in one tail.

---

### 8.

```python
2 * (1 - stats.t.cdf(...))
```

Multiplies by 2 because it is a **two-tailed test**.

This gives the p-value.

---

### 9.

```python
if p_value < alpha:
    print("Reject H0")
else:
    print("Accept H0")
```

Decision rule:

```text
p < α  → Reject H₀
p ≥ α → Fail to Reject H₀
```

---

# Remember

```text
T-statistic  → How far the sample mean is from μ.

Degrees of freedom → Controls the shape of the t-distribution.

P-value → Probability of getting this result by chance.

Critical value → Cutoff obtained from the t-table.

All methods lead to the same final decision.
```